tap_stripe/__init__.py

```diff
@@ -1033,25 +1080,44 @@ def sync_event_updates(stream_name, is_sub_stream):
         # The events stream returns results in descending order, so we
         # cannot bookmark until the entire page is processed
         date_window_start = date_window_end
-        date_window_end = date_window_end + date_window_size
+        date_window_end = date_window_end + events_update_date_window_size
 
-        # Write the parent bookmark value only when the parent is selected
-        if not is_sub_stream:
-            singer.write_bookmark(Context.state,
-                                stream_name + '_events',
-                                'updates_created',
-                                max_created)
-            singer.write_state(Context.state)
+        # Write bookmark for parent or child stream if it is selected
+        write_bookmark_for_event_updates(is_sub_stream, stream_name, sub_stream_name, max_created)
+
+    # max_created is the maximum replication key value among all records.
+    # sync_start_time is epoch time when tap started to sync event updates.
+    # events_update_date_window_size is the size of date_window to make API requests for event updates.
+    # Take back sync_start_time events_update_date_window_size(default 7, maximum 30) days behind to compare
+    # with max_created value. Save a maximum of max_created and sync_start_time because stripe returns
+    # records of the last 30 days for event updates.
+    max_created = max(max_created, sync_start_time - events_update_date_window_size)
+    write_bookmark_for_event_updates(is_sub_stream, stream_name, sub_stream_name, max_created)
 
-        # Write the child bookmark value only when the child is selected
-        if sub_stream_name and Context.is_selected(sub_stream_name):
-            singer.write_bookmark(Context.state,
-                                sub_stream_name + '_events',
-                                'updates_created',
-                                max_created)
-            singer.write_state(Context.state)
     singer.write_state(Context.state)
 
+
+def write_bookmark_for_event_updates(is_sub_stream, stream_name, sub_stream_name, max_created):
+    """
+    Write bookmark for parent and child streams.
+    """
+    # Write the parent bookmark value only when the parent is selected
+    if not is_sub_stream:
+        singer.write_bookmark(Context.state,
+                              stream_name + '_events',
+                              'updates_created',
+                              max_created)
+        singer.write_state(Context.state)
+
+    # Write the child bookmark value only when the child is selected
+    if sub_stream_name and Context.is_selected(sub_stream_name):
+        singer.write_bookmark(Context.state,
+                              sub_stream_name + '_events',
+                              'updates_created',
+                              max_created)
+        singer.write_state(Context.state)
+
+
```

Comment:
Remove this statement and put at the last line of this method.

Reply:
Moved `write_state` to common place.